python/cuda_parallel/cuda/parallel/experimental/_bindings.pyx

```diff
@@ -1618,3 +1618,190 @@ cdef class DeviceUniqueByKeyBuildResult:
                 c_stream
             )
         return status, <object>storage_sz
+
+
+# --------------------------------------------
+#   DeviceUnaryTransform/DeviceBinaryTransform
+# --------------------------------------------
+cdef extern from "cccl/c/transform.h":
+    cdef struct cccl_device_transform_build_result_t:
+        int cc
+        void *cubin
+        size_t cubin_size
+        CUlibrary library
+        CUkernel transform_kernel
+        int loaded_bytes_per_iteration
+
+    cdef CUresult cccl_device_unary_transform_build(
+        cccl_device_transform_build_result_t *build_ptr,
+        cccl_iterator_t d_in,
+        cccl_iterator_t d_out,
+        cccl_op_t op,
+        int, int, const char *, const char *, const char *, const char *
+    ) nogil
+
+    cdef CUresult cccl_device_unary_transform(
+      cccl_device_transform_build_result_t build,
+      cccl_iterator_t d_in,
+      cccl_iterator_t d_out,
+      unsigned long long num_items,
+      cccl_op_t op,
+      CUstream stream) nogil
+
+    cdef CUresult cccl_device_binary_transform_build(
+      cccl_device_transform_build_result_t* build_ptr,
+      cccl_iterator_t d_in1,
+      cccl_iterator_t d_in2,
+      cccl_iterator_t d_out,
+      cccl_op_t op,
+      int, int, const char *, const char *, const char *, const char *
+    ) nogil
+
+    cdef CUresult cccl_device_binary_transform(
+      cccl_device_transform_build_result_t build,
+      cccl_iterator_t d_in1,
+      cccl_iterator_t d_in2,
+      cccl_iterator_t d_out,
+      unsigned long long num_items,
+      cccl_op_t op,
+      CUstream stream) nogil
+
+    cdef CUresult cccl_device_transform_cleanup(
+        cccl_device_transform_build_result_t *build_ptr,
+    ) nogil
+
+
+cdef class DeviceUnaryTransform:
```

Comment:
We could define `DeviceTransformBase` that has `build_data` member and no-arg `__cinit__` that calls `memset(&self.build_data, 0, sizeof(cccl_device_transform_build_result_t))` but does not device `__dealloc__`.

Then `DeviceUnaryTransform` and `DeviceBinaryTransform` would inherit from `DeviceTransformBase`, call `super().__init__()` from its `__cinit__` code and proceed as is.

 Just an idea.